.github/workflows/build-torch-wheels.yml

```
@@ -0,0 +1,173 @@
+name: Build Torch Wheels
+
+on:
+  push:
+    branches: [main]
+    tags: ["*"]
+  pull_request:
+    # Check all PR
+
+#env:
+#  SPHERICART_NO_LOCAL_DEPS: "1"
+
+jobs:
+  build-torch-wheels:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        cibw-arch: ["x86_64"]
+        python-version: ["3.11", "3.12"]
+        pytorch-version: ["2.4.1", "2.5.0"]
+        cuda-version: ["12.4"]
+    env:
+      CIBW_SKIP: cp36-* cp37-* cp38-* cp39-* cp310-*
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+    
+    - name: Github Actions Envs Setup
+      run: |
+        CUVERSION="${{ matrix.cuda-version }}"
+        PYTHONVERSION="${{ matrix.python-version }}"
+
+        CU_VERSION_NO_DOT=${CUVERSION//./}
+        echo CU_VERSION_NO_DOT=${CU_VERSION_NO_DOT} >> $GITHUB_ENV
+
+        CU_VERSION_DASH=${CUVERSION//./-}
+        echo CU_VERSION_DASH=${CU_VERSION_DASH} >> $GITHUB_ENV
+
+        PYTHON_VER_NO_DOT=${PYTHONVERSION//./}
+        echo PYTHON_VER_NO_DOT=${PYTHON_VER_NO_DOT} >> $GITHUB_ENV
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+      with:
+        platforms: all
+  
+    # Build the custom Manylinux Docker image
+    - name: Build Manylinux Docker Image
+      run: |
+        docker build --no-cache \
+        -t manylinux2014_"${{ matrix.cibw-arch }}" \
+        --build-arg PYTHON_VER="${{ matrix.python-version }}" \
+        --build-arg PYTHON_VER_NO_DOT="${{ env.PYTHON_VER_NO_DOT }}" \
+        --build-arg CUDA_VER="${{ matrix.cuda-version }}" \
+        --build-arg CUDA_VER_NO_DOT="${{ env.CU_VERSION_NO_DOT }}" \
+        --build-arg CUDA_VER_DASH="${{ env.CU_VERSION_DASH }}" \
+        --build-arg PYTORCH_VERSION="${{ matrix.pytorch-version }}" \
+        scripts/manylinux2014_"${{ matrix.cibw-arch }}"
+    
+    # Set up Python environment
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "${{ matrix.python-version }}"
+    
+    - name: Install cibuildwheel
+      run: python -m pip install cibuildwheel==2.22.0
+
+    - name: Build torch wheels
+      run: python -m cibuildwheel ./sphericart-torch
+      env:
+        CUDA_HOME: /usr/local/cuda
+        PIP_EXTRA_INDEX_URL: "https://download.pytorch.org/whl/cu${{  env.CUVERSION }}"
+        CIBW_BUILD_VERBOSITY: 3
+        CIBW_BUILD: "cp${{ env.PYTHON_VER_NO_DOT }}-*"
+        CIBW_BUILD_FRONTEND: "pip; args: --no-build-isolation"
```

```suggestion
        CIBW_BUILD_FRONTEND: "pip; args: --no-build-isolation --check-build-dependencies"
```

If we don't use build isolation, we should ensure that the pre-installed dependencies match what the code wants.